TestsFoundation/TestsFoundation/Extensions/XCUIElementExtensions.swift

```diff
@@ -19,50 +19,241 @@
+    func cutText() -> XCUIElement {
+        let selectAll = app.find(label: "Select All")
+        actionUntilElementCondition(action: .tap, element: selectAll, condition: .visible)
+        selectAll.hit()
+        app.find(label: "Cut").hit()
+        return self
+    }
+
+    func relativeCoordinate(x: CGFloat, y: CGFloat) -> XCUICoordinate {
+        return coordinate(withNormalizedOffset: CGVector(dx: x, dy: y))
+    }
+
+    func pullToRefresh(x: CGFloat = 0.5) {
+        relativeCoordinate(x: x, y: 0.2).press(forDuration: 0.05, thenDragTo: relativeCoordinate(x: x, y: 1.0))
+    }
+
+    func tapAt(_ point: CGPoint) {
+        waitUntil(condition: .hittable)
+        coordinate(withNormalizedOffset: .zero).withOffset(CGVector(dx: point.x, dy: point.y)).tap()
+    }
+
+    // MARK: Find
+    func find(label: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(label: label).firstMatch
     }
 
-    func find(labelContaining needle: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(labelContaining: needle).firstElement
+    func find(labelContaining needle: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(labelContaining: needle).firstMatch
     }
 
-    func find(id: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(id: id).firstElement
+    func find(id: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(id: id).firstMatch
     }
 
-    func find(idStartingWith prefix: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(idStartingWith: prefix).firstElement
+    func find(idStartingWith prefix: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(idStartingWith: prefix).firstMatch
     }
 
-    func find(value: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(value: value).firstElement
+    func find(value: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(value: value).firstMatch
     }
 
-    func find(type: XCUIElement.ElementType, index: Int = 0) -> Element {
-        return XCUIElementQueryWrapper(descendants(matching: type), index: index)
+    func find(type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).firstMatch
     }
 
-    func find(id: String, label: String, type: XCUIElement.ElementType = .any) -> Element {
-        return descendants(matching: type).matching(id: id).matching(label: label).firstElement
+    func find(id: String, label: String, type: ElementType = .any) -> XCUIElement {
+        return descendants(matching: type).matching(id: id).matching(label: label).firstMatch
```

Comment:
This is minor but sometimes we have the `return` statement written for such one liners sometimes not. We could unify this within this file.

Reply:
Done.